diff --git a/arch/arm/boot/dts/qcom/kenzo/batterydata-coslight-4000mah-4400mv-10k.dtsi b/arch/arm/boot/dts/qcom/kenzo/batterydata-coslight-4000mah-4400mv-10k.dtsi
index ab2f094..cfeaaf2 100644
--- a/arch/arm/boot/dts/qcom/kenzo/batterydata-coslight-4000mah-4400mv-10k.dtsi
+++ b/arch/arm/boot/dts/qcom/kenzo/batterydata-coslight-4000mah-4400mv-10k.dtsi
@@ -9,7 +9,7 @@ qcom,ct865_coslight_4000mah_10k_averaged_masterslave_oct2nd2015 {
 	qcom,chg-rslow-comp-c2 = <3185429>;
 	qcom,chg-rs-to-rslow = <896437>;
 	qcom,chg-rslow-comp-thr = <0x99>;
-	qcom,fastchg-current-ma = <2000>;
+	qcom,fastchg-current-ma = <2400>;
 	qcom,checksum = <0xA7BE>;
 	qcom,gui-version = "PMI8950GUI - 2.0.0.14";
 	qcom,fg-profile-data = [
diff --git a/arch/arm/boot/dts/qcom/kenzo/batterydata-scud-4000mah-4400mv.dtsi b/arch/arm/boot/dts/qcom/kenzo/batterydata-scud-4000mah-4400mv.dtsi
index 77c12c0..497aedd 100644
--- a/arch/arm/boot/dts/qcom/kenzo/batterydata-scud-4000mah-4400mv.dtsi
+++ b/arch/arm/boot/dts/qcom/kenzo/batterydata-scud-4000mah-4400mv.dtsi
@@ -9,6 +9,7 @@ qcom,longcheer_ct865scud_4000mah_battery {
 	qcom,chg-rslow-comp-c2 = <2592188>;
 	qcom,chg-rs-to-rslow = <1432403>;
 	qcom,chg-rslow-comp-thr = <0x99>;
+	qcom,fastchg-current-ma = <2400>;
 	qcom,checksum = <0x6EE9>;
 	qcom,gui-version = "PMI8950GUI - 2.0.0.15";
 	qcom,fg-profile-data = [
diff --git a/arch/arm/boot/dts/qcom/kenzo/batterydata-sunwoda-4000mah-4400mv.dtsi b/arch/arm/boot/dts/qcom/kenzo/batterydata-sunwoda-4000mah-4400mv.dtsi
index 5ee820f..53e920a 100644
--- a/arch/arm/boot/dts/qcom/kenzo/batterydata-sunwoda-4000mah-4400mv.dtsi
+++ b/arch/arm/boot/dts/qcom/kenzo/batterydata-sunwoda-4000mah-4400mv.dtsi
@@ -9,7 +9,7 @@ qcom,2162464_longcheer_ct865sunwoda_4000mah_averaged_masterslave_oct12th2015 {
 	qcom,chg-rslow-comp-c2 = <3467482>;
 	qcom,chg-rs-to-rslow = <911611>;
 	qcom,chg-rslow-comp-thr = <0x99>;
-	qcom,fastchg-current-ma = <2000>;
+	qcom,fastchg-current-ma = <2400>;
 	qcom,checksum = <0x43DE>;
 	qcom,gui-version = "PMI8950GUI - 2.0.0.14";
 	qcom,fg-profile-data = [
diff --git a/arch/arm/boot/dts/qcom/msm-pmi8950.dtsi b/arch/arm/boot/dts/qcom/msm-pmi8950.dtsi
index 6992bdd..e2b8677 100644
--- a/arch/arm/boot/dts/qcom/msm-pmi8950.dtsi
+++ b/arch/arm/boot/dts/qcom/msm-pmi8950.dtsi
@@ -201,17 +201,18 @@
 			qcom,iterm-ma = <200>;
 			qcom,float-voltage-mv = <4400>;
 			qcom,resume-delta-mv = <200>;
+			qcom,fastchg-current-ma = <2400>;
 			qcom,chg-inhibit-fg;
 			qcom,rparasitic-uohm = <100000>;
 			qcom,bms-psy-name = "bms";
-			qcom,thermal-mitigation = <2000 1500 700 600 0>;
+			qcom,thermal-mitigation = <2400 1800 700 600 0>;
 			qcom,fastchg-current-comp = <900>;
-			qcom,float-voltage-comp = <11>;
 			/*11 for 4.2V, 16 for 4.1V*/
 
 			qcom,force-aicl-rerun;
 			qcom,aicl-rerun-period-s = <180>;
 			qcom,autoadjust-vfloat;
+			qcom,enable-hvdcp-9v;
 			qcom,pmic-revid = <&pmi8950_revid>;
 			qcom,charge-unknown-battery;
 			qcom,board_ntc-vadc = <&pm8950_vadc>;
@@ -313,9 +314,7 @@
 			compatible = "qcom,qpnp-fg";
 			#address-cells = <1>;
 			#size-cells = <1>;
-			qcom,resume-soc = <99>;
-			qcom,fg-delta-soc = <1>;
-			qcom,vbat-estimate-diff-mv = <100>;
+			qcom,resume-soc = <95>;
 			status = "okay";
 			qcom,bcl-lm-threshold-ma = <127>;
 			qcom,bcl-mh-threshold-ma = <405>;
